### PR TITLE
[ITA-327] Add mojo-compatibility stage.

### PR DESCRIPTION
**Pull request by michal-raska**

### CODE REVIEW

File: scripts/jenkins/groovy/defineTestStages.groovy [hunk 1]
```diff
@@ -167,6 +167,11 @@ def call(final pipelineContext) {
       stageName: 'R3.4 Generate Docs', target: 'r-generate-docs-jenkins', archiveFiles: false,
       timeoutValue: 5, component: pipelineContext.getBuildConfig().COMPONENT_R, hasJUnit: false,
       archiveAdditionalFiles: ['h2o-r/h2o-package/**/*.html', 'r-generated-docs.zip']
+    ],
+    [
+      stageName: 'MOJO Compatibility', target: 'test-mojo-compatibility',
+      archiveFiles: false, timeoutValue: 20, , pythonVersion: '3.5', hasJUnit: false,
```
Comment (michalkurka) [inline scripts/jenkins/groovy/defineTestStages.groovy]: I think we can upgrade Python to 3.6

File: scripts/jenkins/Makefile.jenkins [hunk 2]
```diff
@@ -313,4 +313,12 @@ coverage-junit-algos-jenkins:
 coverage-junit-algos:
 	$(call sed_test_scripts)
 	./gradlew h2o-algos:test $$ADDITIONAL_GRADLE_OPTS
-	./gradlew jacocoMergeExecs $$ADDITIONAL_GRADLE_OPTS
\ No newline at end of file
+	./gradlew jacocoMergeExecs $$ADDITIONAL_GRADLE_OPTS
+
+test-mojo-compatibility:
+ifeq ($(wildcard ../mojoland),)
+	git clone https://github.com/h2oai/mojoland ../mojoland
```
Comment (michalkurka) [inline scripts/jenkins/Makefile.jenkins]: I would prefer not to write to the upper-level directory. If "mojoland"  is not found the target should fail and instruct the user to clone it manually.

Jenkins builds should always provide up-to-date checkout in the pipeline itself.

But this solution is fine for now.

File: scripts/jenkins/groovy/defineTestStages.groovy (hunk 1 above)
Comment (michal-raska) [reply scripts/jenkins/groovy/defineTestStages.groovy]: Updated

**Comment by michal-raska** [comment]
@michalkurka mojoland PR is merged, I'll wait for tests here and merge this as well.

**Pull request merged**